### DISCUSSION TITLE
Question regarding the use of mono by .NET Core / 5 and .NET 5 compatibility with the .NET Framework

### DISCUSSION THREAD

**Discussion by Maniues** [dotnet/runtime#50150] [Q&A] [answered]
What is the role of Mono in the dotnet / runtime project?
Why .NET Framework applications cannot run out-of-the-box on .NET Core and .NET 5/6?

**Comment by danmoseley** [comment, marked as answer]
Hi @Maniues questions like this probably belong here in Discussions not as an Issue.

1. Mono is one of the two runtimes that are supported by the .NET 6 project. CoreCLR is the other. They have different tradeoffs. Historically Mono was mostly used on devices as part of Xamarin apps, in the Unity game framework, and more recently it is the runtime used for WASM-based Blazor apps. It was AOT first. CoreCLR historically has been focused at desktop and servers and was JIT first. From .NET 6 onwards, they will run under identical libraries, and will be highly compatible. Over time we expect to find ways to converge parts where it makes sense. For example there are investments this cycle into new AOT capabilities for CoreCLR.

2. Apps (ie., exe's) cannot, because the app model is quite different. .NET Framework apps are designed to cause Windows to load clr.dll, which is the .NET Framework runtime. .NET Core/.NET 5+ have a quite different model for loading dependencies for example. That's apps. Mostly when you want to do reuse you care about libraries (dll's). To some extent, libraries built targeting .NET Framework may run on .NET Core (meaning again, .NET Core 1.0-3.1, and .NET 5+) however this is not robust. If you want to build libraries that can run on either, you should target something called .NET Standard, which is a standard and infrastructure that allows them to do so in a supported way. This is very commonly done, and makes it easier to migrate code piece by piece if that is your goal.

Does this help? There is more info about these things in the docs. Please let me know if you need a pointer ot anything specific.

**Reply by Maniues** [reply]
Yes thank you. It helped.
Sorry for posting this on issues. Next time I'll have a little idea of where to put my question.

**Reply by danmoseley** [reply]
No problem at all.

**Reply by Maniues** [reply]
@danmoseley  One more question came to my mind.
Why is mono used in dotnet / runtime since you have CoreCLR?

**Reply by directhex** [reply]
Porting Mono to use CoreCLR's class library was less work than getting CoreCLR to fit inside an iPhone
